### ISSUE TITLE
All messages pending until restart

### ISSUE THREAD

**Issue by gtmadev** [postalhq/postal#853]
Ever since I started using a Rabbitmq cluster, I see  that all the messages start showing as pending after a while. I will look at Rabbitmq admin though and I see that Postal IS connected. But, it doesn't send messages out until I restart Postal. Any idea which of all these logs I should be looking at to see why it won't send?

The Rabbitmq setup is 3 nodes load balanced behind HAPROXY. I also use the Consul plugin so it knows which nodes are available and HAPROXY will only serve up healthy nodes.

FYI.. I do not have to do anything to Rabbitmq to make things work. I restart postal, and then it works. So it definitely seems like a problem with Postal and its ability to get reconnected to RMQ.

But a valid first step here is to see what the logs are saying. Thx

**Comment by gtmadev** [comment]
UPDATE:

Okay, I did something that I should have known would cause things to break. On my HAPROXY nodes (inside the swarm), I specified some internal DNS servers that resolve names on my private network. This was so it would easier (so I thought) to point a service at a non-swarm private ip. Little did I know that the additional DNS servers specified conflicted with the swarm nodes names. This was causing all kinds of crazy problems with the swarm and all the services published.

I removed the private DNS servers from the HAPROXY nodes, and like magic, everything functioning again rock solid. Another lesson learned with Docker Swarm. DO NOT mess with DNS. Just create host entries in the stack or compose file if you really don't want to hard code the IP of a non swarm server.
